dds_web/commands.py

```diff
@@ -775,75 +783,268 @@ def quarterly_usage():
+        "1. Check the logs on OpenSearch.\n"
+        "2. The DDS team should enter the backend container and run the command `flask send-usage`.\n"
+        "3. Check that you receive a new email indicating that the command was successful.\n"
+    )
+
+    end = current_time()
+    flask.current_app.logger.debug(f"Month now: {end.month}")
+
+    start = end - relativedelta(months=months)
+    flask.current_app.logger.debug(f"Month {months} months ago: {start.month}")
+
+    flask.current_app.logger.debug(f"Start: {start}")
+    flask.current_app.logger.debug(f"End: {end}")
+
+    # CSV files to send
+    csv_file_names = []
+
+    have_failed = False  # Flag to check if any csv files failed to be generated
+
+    # Iterate through units
+    for unit in models.Unit.query:
+        # Generate CSV file name
+        csv_file_name = pathlib.Path(
+            f"{unit.public_id}_Usage_Months-{start.month}-to-{end.month}.csv"
+        )
+        flask.current_app.logger.debug(f"CSV file name: {csv_file_name}")
+
+        # Total usage for unit
+        total_usage = 0
+
+        # Open the csv file
+        try:
+            with csv_file_name.open(mode="w+", newline="") as file:
+                csv_writer = csv.writer(file)
+                csv_writer.writerow(
+                    [
+                        "Project ID",
+                        "Project Title",
+                        "Project Created",
+                        "Time Collected",
+                        "Byte Hours",
+                    ]
+                )
+
+                # Get usage rows connected to unit, that have been collected between X months ago and now
+                for usage_row, project_row in page_query(
+                    db.session.query(models.Usage, models.Project)
+                    .join(models.Project)
+                    .filter(
+                        models.Project.responsible_unit == unit,
+                        models.Usage.time_collected.between(start, end),
+                    )
+                ):
+                    # Increase total unit usage
+                    total_usage += usage_row.usage
+
+                    # Save usage row info to csv file
+                    csv_writer.writerow(
+                        [
+                            project_row.public_id,
+                            project_row.title,
+                            project_row.date_created,
+                            usage_row.time_collected,
+                            usage_row.usage,
+                        ]
+                    )
+
+                # Save total
+                csv_writer.writerow(["--", "--", "--", "--", total_usage])
+        except Exception as e:
+            # Catch exception, dont raise it. So it can continue to next unit
+            flask.current_app.logger.error(f"Error writing to CSV file: {e}")
+            # delete already generated csv files
+            [csv_file.unlink() for csv_file in csv_file_names]
```

Comment:
I think we just want to remove the one file that failed? This removes all
```suggestion
            csv_file_name.unlink()
```
or something?

Reply:
Ah, yes. I forgot to remove it after changing of the function keeping working even after failing with one file

Reply:
We still need the `csv_file_name.unlink()` there though (now it's just deleted), otherwise the csv file won't be appended to the `csv_file_names` list but it also won't be removed from the backend

Reply:
Just checking that you've seen this @rv0lt

Reply:
Yes, I am trying to see how to change it. Because some exceptions would never generate the file (ex, one throw in the open funct or the one I am mocking in the tests)

Reply:
You can do `missing_ok=True`
https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink